File: School-Management-System/WebApi/ClientApp/src/app/home/components/fees/model/studentFeeDetail.viewModel.ts
export interface StudentFeeDetailViewModel {
  id: string;
  academicYearName: string;
  className: string;
  feeType: string;
  feeMonth?: string; // Use string for date from API (ISO format)
  discountAmount: number;
  fineAmount: number;
  totalAmount: number;
  netAmount: number;
  paidAmount: number;
  pendingAmount: number;
  isPaid: boolean;
}

File: School-Management-System/Application/Fees/Dtos/StudentFeeDetailViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Fees.Dtos
{
    public class StudentFeeDetailViewModel
    {
        public string ClassName { get; set; } = string.Empty;
        public string FeeType { get; set; } = string.Empty;
        public DateTime? FeeMonth { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal PaidAmount { get; set; }
        public decimal PendingAmount { get; set; }
        public bool IsPaid { get; set; }
    }
}
